h5py/tests/test_selections.py

```diff
@@ -26,59 +36,113 @@ class TestTypeGeneration(TestCase):
-        out, format = sel.read_dtypes(dt, ())
+        out, format = sel2.read_dtypes(dt, ())
         self.assertEqual(out, format)
         self.assertEqual(out, dt)
 
         # Explicit selection of fields -> requested fields
-        out, format = sel.read_dtypes(dt, ('a','b'))
+        out, format = sel2.read_dtypes(dt, ('a','b'))
         self.assertEqual(out, format)
         self.assertEqual(out, np.dtype( [('a','i'), ('b','f')] ))
 
         # Explicit selection of exactly one field -> no fields
-        out, format = sel.read_dtypes(dt, ('a',))
+        out, format = sel2.read_dtypes(dt, ('a',))
         self.assertEqual(out, np.dtype('i'))
         self.assertEqual(format, np.dtype( [('a','i')] ))
 
+        # Field does not apear in named typed
+        with self.assertRaises(ValueError):
+            out, format = sel2.read_dtypes(dt, ('j', 'k'))
 
-class TestScalarSliceRules(TestCase):
+class TestScalarSliceRules(BaseSelection):
 
     """
         Internal feature: selections rules for scalar datasets
     """
 
-    def setUp(self):
-        self.f = h5py.File(self.mktemp(), 'w')
-        self.dsid = self.f.create_dataset('x', ()).id
-
-    def tearDown(self):
-        if self.f:
-            self.f.close()
-
     def test_args(self):
         """ Permissible arguments for scalar slicing """
-        shape, selection = sel.read_selections_scalar(self.dsid, ())
+        shape, selection = sel2.read_selections_scalar(self.dsid, ())
         self.assertEqual(shape, None)
         self.assertEqual(selection.get_select_npoints(), 1)
 
-        shape, selection = sel.read_selections_scalar(self.dsid, (Ellipsis,))
+        shape, selection = sel2.read_selections_scalar(self.dsid, (Ellipsis,))
         self.assertEqual(shape, ())
         self.assertEqual(selection.get_select_npoints(), 1)
 
         with self.assertRaises(ValueError):
-            shape, selection = sel.read_selections_scalar(self.dsid, (1,))
+            shape, selection = sel2.read_selections_scalar(self.dsid, (1,))
+
+        dsid = self.f.create_dataset('y', (1,)).id
+        with self.assertRaises(RuntimeError):
+            shape, selection = sel2.read_selections_scalar(dsid, (1,))
+
+class TestSelection(BaseSelection):
+
+    """ High-level routes to generate a selection
+    """
+
+    def test_selection(self):
+        dset = self.f.create_dataset('dset', (100,100))
+        regref = dset.regionref[0:100, 0:100]
+
+        # args is numpy.ndarray, return a FancySelection
+        st = sel.select((10,), np.array([1,2,3], dtype='b'), dset)
+        self.assertIsInstance(st, sel.FancySelection)
+
+        self.assertEqual(st.mshape, (3,100))
+        self.assertEqual(st.array_shape, ())
+
+        # expand shape
+        self.assertEqual(st.expand_shape(()), ())
+        with self.assertRaises(TypeError):
+            st.expand_shape((4,100))
+
+        # broadcast in FancySelection
+        st.broadcast((5,100))
```

Comment:
This doesn't actually test anything - `.broadcast()` is a generator, so you need to iterate over it in order to run it.